### PR TITLE
Add missing types for assembly nodes.

### PR DESCRIPTION
**Pull request by Leeleo3x**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -625,11 +675,15 @@ export interface Visitor {
...
+  'Continue:exit'?: (node: Continue) => any;
 }
 export interface ParserOpts {
   tolerant?: boolean;
   range?: boolean;
   loc?: boolean;
 }
 export function parse(sourceCode: string, parserOpts: ParserOpts): ASTNode;
-export function visit(ast: ASTNode, visitor: Visitor): void;
+export function visit(ast: ASTNode | ASTNode[] | null | undefined, visitor: Visitor): void;
```
Comment (federicobond) [inline index.d.ts]: Visit only officially accepts non-null, non-array AST nodes here. The behavior for other cases remains undefined.

Comment (Leeleo3x) [reply index.d.ts]: Should list of ASTNode be accepted?

It will simplify code like
```
visitBlock(node: Block) {
  visit(node.statements, visitor)
}
```

Comment (federicobond) [reply index.d.ts]: Why not just visit the whole Block node in that case? Writing the above kind of code couples the users much more to the internal structure of the AST.

Comment (Leeleo3x) [reply index.d.ts]: I see. Thanks.

**Pull request merged**